api/src/http/routes/change-password.ts

```diff
@@ -23,8 +26,51 @@ export async function changePassword(server: FastifyInstance) {
             id: number;
         };
 
-        const profile = await profileRepository.findById(decoded.id);
+        let changePassword: ChangePasswordInput;
 
-        return reply.send(profile);
+        try {
+            changePassword = changePasswordSchema.parse(request.body);
+        } catch (error) {
+            return reply.status(400).send({
+                message: (error as Error).message,
```

Comment:
É bom fazer isso pra ter uma mensagem de erro melhor.

Tem que importar o `ZodError` do `zod`.

```suggestion
                message: (error as ZodError).issues[0].message,
```

Reply:
Poderia me explicar este issues[0]?

Reply:
Todo erro de validação com o zod retorna um objeto. Esse objeto tem uma key `issues`. O trecho `(error as ZodError).issues` acessa essa key.

`issues` é um array contendo todos os erros de validação que ocorreram. A gente quer pegar apenas o primeiro erro. Fazemos isso com `(error as ZodErro).issues[0]`.

Cada posição desse array é um objeto que contém o várias keys. Dentre elas há a key `message` que tem a mensagem configurada lá no schema do zod.

Logo `(error as ZodError).issues[0].message` retorna a mensagem de erro da primeira validação que falhou e que está listada no array `issues`.

Reply:
Putz, bem cheio de zig zag

Reply:
Entendi